hosts/galileo/__init__.py

```diff
@@ -0,0 +1,395 @@
+################################################################################
+# hosts/galileo/__init__.py: Galileo class
+#
+# Utility functions for managing SPICE kernels while working with Galileo data
+# sets.
+################################################################################
+import numpy as np
+import unittest
+import os.path
+
+import julian
+import textkernel
+import spicedb
+import cspyce
+import oops
+
+from oops.body import Body
+
+TOUR = (2003 - 2000) * 365 * 86400      # Rough ET dividing Saturn from Jupiter
+
+################################################################################
+# Routines for managing the loading of C and SP kernels
+################################################################################
+
+# Make sure the leap seconds have been loaded
+oops.spice.load_leap_seconds()
+
+# We load CK and SPK files on a very rough month-by-month basis. This is simpler
+# than a more granular approach involving detailed calendar calculations. We
+# divide the period October 1, 1997 to October 1, 2017 up into 240 "months" of
+# equal length. Given any TDB, we quickly determine the month within which it
+# falls. Each month is associated with a list of kernels that should be loaded
+# whenever information is needed about any time within that month +/- 12 hours.
+# The kernels needed for a given month only get loaded when they are needed, and
+# are only loaded once. For any geometry calculation involving Galileo, a quick
+# call to load_ck(time) or load_spk(time) will ensure that the information is
+# available.
+
+################################################################################
+
+class Galileo(object):
+    """An instance-free class to hold Galileo-specific parameters."""
+
+    START_TIME = '1989-10-18'
+    STOP_TIME  = '2003-09-21'
+    MONTHS = 240        # 20 years * 12 months/year
+
+    TDB0 = julian.tdb_from_tai(julian.tai_from_iso(START_TIME))
+    TDB1 = julian.tdb_from_tai(julian.tai_from_iso(STOP_TIME))
+    DTDB = (TDB1 - TDB0) / MONTHS
+    SLOP = 43200.
+
+    CK_LOADED = np.zeros(MONTHS, dtype='bool')      # True if month was loaded
+    CK_LIST   = np.empty(MONTHS, dtype='object')    # Kernels needed by month
+    CK_DICT   = {}      # Dictionary keyed by filespec returns kernel info
+                        # object, but only if loaded.
+
+    SPK_LOADED  = np.zeros(MONTHS, dtype='bool')
+    SPK_LIST    = np.empty(MONTHS, dtype='object')
+    SPK_DICT    = {}
+
+    loaded_instruments = []
+
+    initialized = False
+
+    ############################################################################
+
+    @staticmethod
+    def initialize(ck='reconstructed', planets=None, asof=None,
```

Comment:
Galileo doesn't have reconstructed vs. predicted kernels, or gapfill kernels. We don't have these parameters for Voyager, so they should be removed here.

Reply:
True, but also probably not worth worrying about right now.